middleware/auth.ts

```diff
@@ -0,0 +1,31 @@
+import { Request, Response, NextFunction } from "express";
+import admin from "firebase-admin";
+import { StatusCodes } from "http-status-codes";
+
+admin.initializeApp({
+  credential: admin.credential.applicationDefault(),
+});
+
+export const authenticateFirebaseToken = async (req: Request, res: Response, next: NextFunction) => {
+  const authHeader = req.headers.authorization;
+
+  if (!authHeader || !authHeader.startsWith("Bearer ")) {
+    return res
+      .status(StatusCodes.UNAUTHORIZED)
+      .json({ message: "Authorization 헤더가 누락되었거나 잘못된 형식입니다." });
+  }
+
+  const idToken = authHeader.split("Bearer ")[1];
```

Comment:
아래와 같이 표현해서 매직넘버를 없애고 의도를 조금 더 드러내는 것도 좋을 것 같아요!
```suggestion
  const [, token] = authHeader.split("Bearer ");
```